ast/ast.go

```diff
@@ -2878,6 +2884,76 @@ type Analyze struct {
 	Analyze token.Pos // position of "ANALYZE" keyword
 }
 
+// CreateModelColumn is a single column definition node in CREATE MODEL.
+//
+//	{{.Name | sql}} {{.DataType | sql}} {{.Options | sqlOpt}}
+type CreateModelColumn struct {
+	// pos = Name.pos
+	// end = (Options ?? DataType).end
+
+	Name     *Ident
+	DataType SchemaType
+	Options  *Options // optional
+}
+
+// CreateModelInputOutput is INPUT and OUTPUT column list node.
+//
+//	INPUT ({{.InputColumns | sqlJoin ", "}}) OUTPUT ({{.OutputColumns | sqlJoin ", "}})
+type CreateModelInputOutput struct {
+	// pos = Input
+	// end = Rparen + 1
+
+	Input  token.Pos
+	Rparen token.Pos // position of the last ")"
+
+	InputColumns  []*CreateModelColumn
+	OutputColumns []*CreateModelColumn
+}
+
+// CreateModel is CREATE MODEL statement node.
+//
+//	CREATE {{if .OrReplace}}OR REPLACE{{end}} MODEL {{if .IfNotExists}}IF NOT EXISTS{{end}} {{.Name | sql}}
+//	{{.InputOutput | sqlOpt}}
+//	REMOTE
+//	{{.Options | sqlOpt}}
+type CreateModel struct {
+	// pos = Create
+	// end = Options.end || Remote + 6
+
+	Create      token.Pos // position of "CREATE" keyword
+	Remote      token.Pos // position of "REMOTE" keyword
+	OrReplace   bool
```

Comment:
Insert a newline here. (Sorry, I didn't use "suggestion" because I guess it breaks `go fmt` format.).